test/lib/code_corps_web/controllers/project_user_controller_test.exs

```diff
@@ -99,6 +99,8 @@ defmodule CodeCorpsWeb.ProjectUserControllerTest do
         |> CodeCorps.Repo.preload([:project, :user])
         |> CodeCorps.Emails.ProjectUserAcceptanceEmail.create()
 
+      email |> CodeCorps.Mailer.deliver_now()
+
```

Comment:
Test is not passing without the explicit call to `deliver_now`. It used to pass earlier without this. Any ideas why?

Reply:
The email should have been delivered from the controller, so you should definitely remove this line. Instead, if you look at the `schedule_email` plug, it calls `maybe_send_email`, which then uses the changeset to determine if the `project_user` was promoted from "pending", to "contributor" and sends an email in that case.

What we should do instead is, within the action method of the controller, figure out if there was a role change from "pending" to "contributor" and explicitly send the email from the `with` clause in that case.

My suggestion here would be to rename the second `project_user` into `updated_project_user` and then compare the role of the one you initially loaded from the database against the updated one and send an email if there was the change we need.